scripts/irods/test/test_irm.py

```diff
@@ -43,3 +44,37 @@ def test_irm_delete_collection_with_ampersand_in_name_causes_error__issue_3398(s
         self.admin.assert_icommand("irm -r '{0}'".format(collection))
         self.admin.assert_icommand("ils -l '{0}'".format(self.admin.session_collection_trash), 'STDOUT', collection)
 
+    def test_irm_unregister_as_rodsuser(self):
+        filename = 'test_irm_unregister_as_rodsuser'
+        local_data_path = str(os.path.join(self.admin.local_session_dir, filename))
+        lib.make_file(local_data_path, 1024)
+
+        try:
+            # Put a file in rodsuser's home collection
+            # The user should not be able to unregister from the vault
+            self.user.assert_icommand(['iput', local_data_path])
+            data_in_vault_path = str(os.path.join(self.user.get_vault_session_path(), filename))
+            logical_path_to_obj = str(os.path.join(self.user.session_collection, filename))
+            self.user.assert_icommand(['irm', '-U', logical_path_to_obj], 'STDERR', 'CANT_UNREG_IN_VAULT_FILE')
+            self.user.assert_icommand(['ils', '-L', logical_path_to_obj], 'STDOUT', data_in_vault_path)
+            # Unregister from the vault as admin
+            self.admin.assert_icommand(['ichmod', '-M', 'own', self.admin.username, logical_path_to_obj])
+            self.admin.assert_icommand(['irm', '-U', logical_path_to_obj])
+            self.user.assert_icommand(['ils', '-L', logical_path_to_obj], 'STDERR', 'does not exist')
+            self.assertTrue(os.path.exists(data_in_vault_path), msg='Data missing after unregister in [{}]'.format(data_in_vault_path))
+
+            # Register a file in rodsuser's home collection and give write-only access
```

Comment:
write... isn't write-only.   it's 'read-write', but doesn't allow removal/unregistration.

Reply:
I have updated the comment from "write-only" to "only read/write"